File: C-Sharp/Uninstall_app/Uninstall_app.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using RestSharp;
using RestSharp.Authenticators;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;

namespace Uninstall_app
{
    class Uninstall_app
    {
        public static string baseurl = "https://suremdm.42gears.com/api";
        private static string Username = "Username";
        private static string Password = "Password";
        private static string ApiKey = "Your ApiKey";
        static void Main(string[] args)
        {

            // Retrieve Device ID
            string DeviceID = GetDeviceID("Device_Name");
            if (DeviceID != null)
            {
                // Uninstall app
                string status = UninstallApplication(DeviceID, "Application_Name");
                Console.WriteLine(status);
            }
            else
            {
                Console.WriteLine("Device not found!");
            }
            Console.ReadKey();
        }

        private static string UninstallApplication(string deviceID, string appName)
        {
            /* Retrieve AppID (Package name) of an app
               Endpoint: /dynamicjob
                 Method: POST
                 Request Body:
                    {
                        "DeviceID": string,
                        "JobType": string,
                        "PayLoad": string
                    }
                 Authentication:
                     Basic authentication
                 Headers:
                     ApiKey: “Your Api-Key”
              */

            string URL = baseurl + "/dynamicjob";
            var client = new RestClient(URL);
            // Basic authentication header
            client.Authenticator = new HttpBasicAuthenticator(Username, Password);
            // ApiKey Header
            client.AddDefaultHeader("ApiKey", ApiKey);
            // Set content type
            client.AddDefaultHeader("Content-Type", "application/json");
            // Set request method
            var request = new RestRequest(Method.POST);
            // Create job specific PayLoad
            var PayLoad = new
            {
                AppIds = new string[] { GetAppID(deviceID, appName) }
            };
            // convert payload to base64 string
            var PayLoadbytes = System.Text.Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(PayLoad));
            var PayLoadBase64 = System.Convert.ToBase64String(PayLoadbytes);
            // Request payload for refreshing device
            var RequestPayLoad = new
            {
                JobType = "UNINSTALL_APPLICATION",
                DeviceID = deviceID,
                PayLoad = PayLoadBase64
            };
            // Add request body
            request.AddJsonBody(RequestPayLoad);
            // Execute request
            IRestResponse response = client.Execute(request);

            return response.Content.ToString();
        }

        private static string GetAppID(string deviceID, string appName)
        {
            /* Retrieve AppID (Package name) of an app
               Endpoint: /installedapp/android/{DeviceId}/device
                 Method: GET
                 Path parameters:
                     DeviceID: ID of the device allocated by the SureMDM.
                 Authentication:
                     Basic authentication
                 Headers:
                     ApiKey: “Your Api-Key”
              */
            string URL = baseurl + "/installedapp/android/" + deviceID + "/device";
            var client = new RestClient(URL);
            // Basic authentication header
            client.Authenticator = new HttpBasicAuthenticator(Username, Password);
            // ApiKey Header
            client.AddDefaultHeader("ApiKey", ApiKey);
            // Set content type
            client.AddDefaultHeader("Content-Type", "application/json");
            // Set request method
            var request = new RestRequest(Method.GET);
            // Execute request
            IRestResponse response = client.Execute(request);
            if (response.StatusCode == HttpStatusCode.OK)
            {
                var OutPut = JsonConvert.DeserializeObject(response.Content);
                List<JObject> jsonResponse = JsonConvert.DeserializeObject<List<JObject>>(response.Content);
                var a = jsonResponse.Any();
                if ((jsonResponse != null) && (jsonResponse.Any()))
                {
                    foreach (var app in jsonResponse)
                    {
                        if (app.GetValue("Name").ToString().ToUpper() == appName.ToUpper())
                        {
                            return app.GetValue("Id").ToString();
                        }
                    }
                }
            }
            return null;
        }

        private static string GetDeviceID(string deviceName)
        {
            /*  Retreiving information of device
                 Endpoint: /device:
                 Method: POST
                 Request Body:
                     {  
                        "ID": string,
	                    "SearchValue":string,
	                    "Limit":integer,
	                    "SortColumn":string,
	                    "SortOrder":string,
	                    "IsSearch":boolean,
	                    "SearchColumns":string[]
                    }
                 Authentication:
                     Basic authentication
                 Headers:
                     ApiKey: “Your Api-Key” 
             */

            string URL = baseurl + "/device";
            var client = new RestClient(URL);
            //  Basic authentication header
            client.Authenticator = new HttpBasicAuthenticator(Username, Password);
            //  ApiKey Header
            client.AddDefaultHeader("ApiKey", ApiKey);
            //  Set content type
            client.AddDefaultHeader("Content-Type", "application/json");
            //  Set request method
            var request = new RestRequest(Method.POST);
            //  Request payload
            var RequestPayLoad = new
            {
                ID = "AllDevices",
                IsSearch = true,
                Limit = 10,
                SearchColumns = new string[] { "DeviceName" },
                SearchValue = deviceName,
                SortColumn = "LastTimeStamp",
                SortOrder = "asc"
            };
            request.AddJsonBody(RequestPayLoad);

            //  Execute request
            IRestResponse response = client.Execute(request);

            if (response.StatusCode == HttpStatusCode.OK)
            {
                var OutPut = JsonConvert.DeserializeObject<JObject>(response.Content);
                foreach (var device in OutPut["rows"])
                {
                    if ((string)device["DeviceName"] == deviceName)
                    {
                        return device["DeviceID"].ToString();
                    }
                }
            }
            return null;
        }
    }
}

File: Python/Uninstall_app.py
import requests,json,base64

BaseURL="https://suremdm.42gears.com/api"
Username="Username"
Password="Password"
ApiKey="Your ApiKey"

# Method for getting device info
def GetDeviceID(deviceName):
    '''
    Retreiving information of device
        Endpoint: /device
        Method: POST
        Request Body:
            {  
                "ID": string,
                "SearchValue":string,
                "Limit":integer,
                "SortColumn":string,
                "SortOrder":string,
                "IsSearch":boolean,
                "SearchColumns":string[]
            }
        Authentication:
            Basic authentication
        Headers:
            ApiKey: “Your Api-Key”
    '''
    # Api url
    url = BaseURL+"/device"
    # Headers
    headers = {
        # Api-Key header
        'ApiKey': ApiKey,
        # Set Content type
        'Content-Type': "application/json",
        }
    # Basic authentication credentials
    Credentials=(Username,Password)
    # Request body
    PayLoad={
        "ID" : "AllDevices",
        "IsSearch" : bool(1),
        "Limit" : 10,
        "SearchColumns" : ["DeviceName"],
        "SearchValue" : deviceName,
        "SortColumn" : "LastTimeStamp",
        "SortOrder" : "asc"
    }
    # Executing request
    response = requests.post(url,auth=Credentials,json=PayLoad,headers=headers)
    # Extracting required GroupID
    if response.status_code == 200:
        if response.text != '[]':
            data = response.json()
            for group in data['rows']:
                if group['DeviceName'] == deviceName:
                    return group["DeviceID"]
    else:
        return None

def GetAppID(deviceID,appName):
    """
    Retrieve AppID (Package name) of a app
        Endpoint: /installedapp/android/{DeviceId}/device
        Method: GET
        Path parameters:
            DeviceID: ID of the device allocated by the SureMDM.
        Authentication:
            Basic authentication
        Headers:
            ApiKey: “Your Api-Key”
    """
    url = BaseURL+"/installedapp/android/"+deviceID+"/device"
    #  Headers
    headers = {
        #  Api-Key header
        'ApiKey': ApiKey,
        #  Set Content type
        'Content-Type': "application/json",
        }
    #  Basic authentication credentials
    Credentials=(Username,Password)
    #  send request
    response = requests.get(url,auth=Credentials,headers=headers)
    # Extracting required GroupID
    if response.status_code == 200:
        if response.text != '[]':
            jsonResponse = response.json()
            for app in jsonResponse:
                if app['Name'] == appName:
                    return app['Id']
    else:
        return None

def UninstallApp(deviceID,appName):
    """
    Retrieve AppID (Package name) of a app
        Endpoint: /dynamicjob
        Method: POST
        Request Body:
        Type: application/json
        Body parameters:
            DeviceID: ID of the device allocated by the SureMDM.
            JobType: Job type
            PayLoad: Job specific payload
        Authentication:
            Basic authentication
        Headers:
            ApiKey: “Your Api-Key”
    """
    url = BaseURL+"/dynamicjob"
    #  Headers
    headers = {
        #  Api-Key header
        'ApiKey': ApiKey,
        #  Set Content type
        'Content-Type': "application/json",
        }
    #  Basic authentication credentials
    Credentials=(Username,Password)
    # Create job specific PayLoad
    PayLoad={
        "AppIds":[GetAppID(deviceID,appName)]
    }
    # Convert payload to base64 string
    PayLoadStr=json.dumps(PayLoad)
    PayLoadBase64=base64.b64encode(PayLoadStr.encode('utf-8'))
    # Request payload for refreshing device
    RequestPayLoad={
        "JobType": "UNINSTALL_APPLICATION",
        "DeviceID": deviceID,
        "PayLoad":PayLoadBase64.decode('utf-8')
    }
    #  send request
    response = requests.post(url,auth=Credentials,json=RequestPayLoad,headers=headers)
    return response.text

# Retrieve Device ID
DeviceID=GetDeviceID("Device_Name")
if DeviceID!=None:
    # Uninstall app
    status=UninstallApp(DeviceID,"Application_Name")
    print(status)
else:
    print('Device not found!')